libraries/c_sdk/standard/ble/src/services/mqtt_ble/iot_ble_mqtt_transport.c

```diff
@@ -761,74 +870,76 @@ static MQTTStatus_t handleIncomingPublish( MQTTPacketInfo_t * packet,
 
     if( status == MQTTSuccess )
     {
-        status = transportSerializePublish( pBuffer, &publishInfo, packetIdentifier );
+        status = transportSerializePublish( streamBuffer, &publishInfo, packetIdentifier );
     }
 
     return status;
 }
 
 
-static MQTTStatus_t handleIncomingSuback( MQTTPacketInfo_t * packet,
-                                          MQTTFixedBuffer_t * pBuffer )
+static MQTTStatus_t handleIncomingSuback( StreamBufferHandle_t streamBuffer,
+                                          MQTTPacketInfo_t * packet )
 {
     MQTTStatus_t status = MQTTSuccess;
     uint16_t packetIdentifier = 0;
+    uint8_t buffer[ SIZE_OF_SUB_ACK ] = { 0 };
 
     LogDebug( ( "Processing incoming SUBACK from channel." ) );
     status = IotBleMqtt_DeserializeSuback( packet, &packetIdentifier );
 
     if( status == MQTTSuccess )
     {
-        status = transportSerializeSuback( pBuffer, packet->type, packetIdentifier );
-    }
+        buffer[ 0 ] = packet->type;
+        buffer[ 1 ] = 3;
```

Comment:
May be a future change, can MACROs be used instead of hardcoded numbers?

Reply:
Also, SUBACK doesn't always have a fixed remaining length of 3, please refer to the spec here http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718068

Reply:
The companion BLE app sends only one status code (succeeded or failed)  for all of the subscriptions  together. Added comments for this in the code.



> 
> 
> May be a future change, can MACROs be used instead of hardcoded numbers?

Changed to macros and added comments.

Reply:
I see. This may have a problem with the new demos and integration tests as it expects suback status for all the subscriptions. We may want to check these if it will create an issue

Reply:
MQTT demo for BLE takes care of this,  but I will revisit this once the integration test is integrated to work with BLE and follow up in a different PR.